File: src/main/java/pl/androidcoder/solid/lsp/collection/LimitedStorage.java
package pl.androidcoder.solid.lsp.collection;

import java.util.ArrayList;

public class LimitedStorage<T> implements Storage<T> {

    private int limit;

    private ArrayList<T> list = new ArrayList<>();

    public LimitedStorage(int itemLimit){
        limit = itemLimit;
    }

    @Override
    public boolean addItem(T item) {
        if(list.size() >= limit)
            throw new LimitExceededException();
        else
            list.add(item);
        return true;
    }
}

File: src/main/java/pl/androidcoder/solid/lsp/collectionkotlin/LimitedStorage.kt
package pl.androidcoder.solid.lsp.collectionkotlin

import pl.androidcoder.solid.lsp.collection.Storage
import java.util.*

class LimitedStorage<T>(private val limit: Int) : Storage<T> {

    private val list = ArrayList<T>()

    override fun addItem(item: T): Boolean {
        if (list.size >= limit)
            throw LimitExceededException()
        else
            list.add(item)
        return true
    }
}